src/LiquidCrystal.cpp

```diff
@@ -300,6 +310,35 @@ void LiquidCrystal::send(uint8_t value, uint8_t mode) {
   }
 }
 
+int LiquidCrystal::read(bool mode) {
+  if (_rw_pin == 255) { // if there is a RW pin indicated, only then we can read
+    return -1;
+  }
+  else{
+    digitalWrite(_rw_pin, HIGH);//for read
+    digitalWrite(_rs_pin, mode);
+
+    for (int i=0; i<((_displayfunction & LCD_8BITMODE) ? 8 : 4); ++i)
+    {
+      pinMode(_data_pins[i], INPUT);//library initializes data pins in output mode, so we need to change mode for reading
+    }
+
+    uint8_t address;
+    if (_displayfunction & LCD_8BITMODE) {
+      address = read8bits(); 
+    } else {
+      address = read4bits();
+    }
+
+     for (int i=0; i<((_displayfunction & LCD_8BITMODE) ? 8 : 4); ++i)
```

Comment:
```suggestion
    for (int i=0; i<((_displayfunction & LCD_8BITMODE) ? 8 : 4); ++i)
```
Fix indentation

Reply:
Thanks again!
Indentation is fixed now.